pkg/datasources/database_roles.go

```diff
@@ -1,11 +1,11 @@
 package datasources
 
 import (
+	"context"
 	"database/sql"
-	"errors"
 	"log"
 
-	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/snowflake"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk"
```

Comment:
```
"database": {
		Type:        schema.TypeString,
		Optional:    true,
		Description: "The database from which to return the database roles from.",
	},
```

this should not be optional, it should be required.

Reply:
I have changed it to Required true.

But one thing that worries me here is that this is a breaking change in theory (in practice omitting database must have resulted in a runtime error, because it was always needed underneath).